docs/pages/zkvm/sdk-benchmarking.mdx

```diff
@@ -0,0 +1,182 @@
+---
+title: "Nexus zkVM SDK Benchmark"
+lang: en-US
+description: "Benchmark your Nexus project using the SDK"
+image: "/nexus-head.png"
+---
+
+import Image from "next/image";
+
+import { Steps } from "nextra/components";
+
+## Reproducible Proving and Verification
+
+<Steps>
+
+This example demonstrates how to use the Nexus SDK to generate proving and verification keys, compile a guest program, prove its execution, and verify the proof. The example uses the `guest` package as an example guest program.
+
+The `src/main.rs` at host side is as follows:
+
+```rust
+use nexus_sdk::{
+    compile::CompileOpts,
+    nova::seq::{Generate, Nova, PP},
+    Local, Prover, Verifiable,
+};
+
+const PACKAGE: &str = "guest";
+
+fn generate_pp_and_compile() -> (PP, Nova<Local>) {
+    let pp: PP = PP::generate().expect("failed to generate parameters");
+
+    let mut opts = CompileOpts::new(PACKAGE);
+    opts.set_memlimit(8); // use an 8mb memory
+
+    let prover: Nova<Local> = Nova::compile(&opts).expect("failed to compile guest program");
+
+    (pp, prover)
+}
+
+fn prove_execution(pp: &PP, prover: Nova<Local>) -> nexus_sdk::nova::seq::Proof {
+    let proof: nexus_sdk::nova::seq::Proof = prover.prove(pp).expect("failed to prove program");
+
+    proof
+}
+
+fn verify_execution(pp: &PP, proof: &nexus_sdk::nova::seq::Proof) {
+    proof.verify(pp).expect("failed to verify proof");
+}
+
+fn main() {
+    use std::time::Instant;
+
+    let start = Instant::now();
+    let (pp, prover) = generate_pp_and_compile();
+    let duration = start.elapsed();
+    println!(
+        "Time taken to generate PP and compile: {:.2} seconds",
+        duration.as_secs_f64()
+    );
+
+    let start = Instant::now();
+    let proof = prove_execution(&pp, prover);
+    let duration = start.elapsed();
+    println!(
+        "Time taken to prove execution: {:.2} seconds",
+        duration.as_secs_f64()
+    );
+
+    let start = Instant::now();
+    verify_execution(&pp, &proof);
+    let duration = start.elapsed();
+    println!(
+        "Time taken to verify execution: {:.2} seconds",
+        duration.as_secs_f64()
+    );
+}
+```
+
+The `src/guest/src/main.rs` at guest side is as follows:
```

Comment:
An explanation about `#[nexus_rt::profile]` helps readers who are interested in benchmarking.

Reply:
updated in [36f79a4](https://github.com/nexus-xyz/nexus-zkvm/pull/271/commits/36f79a4ffc509fe44f4c288746c674652d698a4b)